Automattic/wp-calypso
# Rename Launchpad to LaunchpadInternal, DefaultWiredLaunchpad to Launchpad

Related to 
* #83303

## Proposed Changes

* To make it much clearer which component developers should be using, this PR renames the existing `Launchpad` component to `LaunchpadInternal` and the recently added `DefaultWiredLaunchpad` to `Launchpad`. As flagged in #83303, most use cases moving forward should use the newer component to avoid unnecessary boilerplate code _and_ to make sure they're leveraging many of the core launchpad features.
* The PR also updates the Launchpad readme to reflect the rename and the existence of the internal component.

## Testing Instructions


This change needs a relatively general smoke test of various launchpad interactions, though there should be no change in behaviour.

* Run this branch locally or via Calypso.live
* For a site that is showing the fullscreen launchpad, open the fullscreen launchpad and verify that the tasks are still working
* For a VideoPress site created via the `/setup/videopress` flow that is still showing the fullscreen launchpad, verify that the tasks etc behave as before
* For a site where you skipped the fullscreen launchpad, verify that the task list in Customer Home is still working
* For a site where we are showing a post-launch launchpad in Customer Home (e.g. newsletter or write intent) verify that the task list is working as expected
* For a site where we are showing a launchpad in Customer Home, add `?flags=launchpad/navigator` to the URL, and verify that the Launchpad Navigator is still working as before.

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
